operators/pkg/utils/network.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package utils collects all the logic shared between different controllers
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// HTTPGet performs a HTTP GET request to the given URL.
+func HTTPGet(ctx context.Context, url string, timeout time.Duration) (statusCode int, contents []byte, err error) {
+	now := time.Now()
+	log := ctrl.LoggerFrom(ctx, "url", url).V(LogDebugLevel).WithName("http-get")
+	defer func() {
+		log.Info("terminated", "duration", time.Since(now))
+	}()
```

Comment:
It is probably more appropriate to create a sub span of  the tracer here (you can create a different LogIfLong function if necessary)